structuraldesign/design/beam_design.py

```
@@ -6,41 +6,147 @@
+
+        # TODO: I don't like those hard numbers
+        if 0.8 * x < hf: #neutral_line at flange (f)
+
+            Rcd_f = 0.8 * 0.85 * self.beam.material.fcd * 1000.0 * x * bf
+            Rcd_w = 0
+            Rcd = Rcd_f + Rcd_w
+
+            Mcd_f = Rcd_f * (d - 0.4 * x)
+
+        else: #neutral_line at web (w)
+
+            Rcd_f = 0.85 * self.beam.material.fcd * 1000.0 * hf * (bf - b)
+
+            Mcd_f = Rcd_f * (d - hf/2.0)
+            Mcd_w = Msd - Mcd_f
+
+            x = _neutral_line(self, Mcd_w, b)
+
+            Rcd_w = 0.85 * self.beam.material.fcd * 1000.0 * b * 0.8 * x
+            Rcd = Rcd_f + Rcd_w
+
+        Mcd_w = Rcd_w * (d - 0.4 * x)
+
+        Mcd = Mcd_f + Mcd_w
+
+        Msd_neg = Msd - Mcd
+
+        Rsd_neg = Msd_neg / (d - dl_neg) #Axial reaction at negative reinforcement
+        Rsd_pos = Rsd_neg + Rcd - Nd #Axial reaction at positive reinforcement
+
+        xd = x / d
+
+        As_pos = Rsd_pos / self.reinforcement.fyd * 10 #to cm2
+
+        if calc_As_min is True:
+            As_min = _minimun_reinforcement(self, calc_As_min)
+            As_pos = max(As_pos, As_min)
+
+        As_neg = Rsd_neg / self.reinforcement.fyd * 10
+
+        return {"neutral_line": x,
+                "xd": xd,
+                "As_pos": As_pos,
+                "As_neg": As_neg
+                }
+
+    def shear_design(self, Vd, strut_angle = None):
+        """strut_angle in degrees"""
+
+        b = self.beam.section.b
+        d = self.beam.section.d
+        fcd = self.beam.material.fcd
+        alpha_v2 = self.beam.material.alpha_v2
+        fyd = self.reinforcement.fyd
+
+        #minimum reinforcement
+        ro_min = 0.2 * self.beam.material.fctm / self.reinforcement.fyk
+        Asw_min = ro_min * b * 10000 #cm2 / m
+
+        Vc0 = 0.6 * self.beam.material.fctd * 1000 * b * d
+
+        if strut_angle is None: #model I
+            Vrd2 = 0.27 * alpha_v2 * fcd * 1000 * b * d
+            #raise some error if Vd > Vrd2 ?
```

What do you think? Should we raise an error if Vd > Vrd2 or just some kind of warning?